test/yaml_edit/append_test.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:pub/src/yaml_edit/yaml_edit.dart';
+import 'package:test/test.dart';
+
+import 'test_utils.dart';
+
+void main() {
+  group('throws PathError', () {
+    test('if it is a map', () {
+      final doc = YamlEditor('a:1');
+      expect(() => doc.appendToList([], 4), throwsPathError);
+    });
+
+    test('if it is a scalar', () {
+      final doc = YamlEditor('1');
+      expect(() => doc.appendToList([], 4), throwsPathError);
+    });
+  });
+
+  group('block list', () {
```

Comment:
Just curious, wouldn't it be easier to do this was golden tests?

Not that I mind having both :D

Reply:
yeah... I'm still a little torn between understanding the division as to what tests I should include as golden tests and what tests I should be including here instead - do you have any tips?

Reply:
I see pros/cons in both directions... I guess golden tests are more useful for large complicated mutations.